### PR TITLE
Add MemoryOfflineStore

### PR DESCRIPTION
**Pull request by simba-git**

### CODE REVIEW

File: ff/serving/provider/offline.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+type ResourceRecord struct {
+	Entity string
+	Value  interface{}
+	TS     time.Time
+}
+
+func (rec ResourceRecord) Check() error {
+	if rec.Entity == "" {
+		return errors.New("ResourceRecord must have Entity set.")
```
Comment (Sami1309) [inline ff/serving/provider/offline.go:48]: should it also have a timestamp?

Comment (Sami1309) [reply ff/serving/provider/offline.go:48]: Also should every new table assert that the Value of of each ResourceRecord it contains be the same type?

File: ff/serving/provider/offline_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	invalidIds := []ResourceID{
+		{Type: Feature},
+		{Name: uuid.NewString()},
+	}
+	for _, id := range invalidIds {
+		if _, err := store.CreateResourceTable(id); err == nil {
+			t.Fatalf("Succeeded in creating invalid ResourceID: %v", id)
+		}
+	}
+}
```
Comment (Sami1309) [inline ff/serving/provider/offline_test.go:102]: no tests for write or resourcerecord check

Comment (simba-git) [reply ff/serving/provider/offline_test.go:102]: Coming in another PR, currently you can only write to an OfflineStore, you can't read

File: ff/serving/provider/offline.go (hunk 1 above)
Comment (simba-git) [reply ff/serving/provider/offline.go:48]: It should have a TS, good catch, the value asserts should come in with Schema which hasn't been implemented yet

**Pull request merged**